scripts/scr_get_item_names/scr_get_item_names.gml

```diff
@@ -0,0 +1,806 @@
+                    if (!equip_data.has_tag(_required_tags[_t])) {
+                        valid = false;
+                        break;
+                    }
+                }
+            }
+            if (_excluded_tags != undefined) {
+                for (var _t = 0; _t < array_length(_excluded_tags); _t++) {
+                    if (equip_data.has_tag(_excluded_tags[_t])) {
+                        valid = false;
+                        break;
+                    }
+                }
+            }
+			if (valid) {
+				array_push(_item_names, equip_data.name);
+			}
+        }
+    }
+    return _item_names;
+}
+
+enum eEQUIPMENT_TYPE {
+    None,
+    PrimaryWeapon = 1,  // LeftHand, Turret, Front, Primary
+    SecondaryWeapon = 2,  // RightHand, Sponson, Secondary
+    Armour = 3,
+    GearUpgrade = 4,
+    MobilityAccessory = 5
+}
+
+enum eEQUIPMENT_SUBTYPE {
+    None,
+    Ranged = 1, // Regular land raider weapons
+    Melee = 2 // Relic land raider weapons
+}
+
+enum eUNIT_TYPE {
+    None,
+    Infantry = 1,
+    Dreadnought = 6,
+    LandRaider = 50,
+    Rhino = 51,
+    Predator = 52,
+    LandSpeeder = 53,
+    Whirlwind = 54
+}
+
+// slot names differ by unit type then equipment type
+// HACK: GM doesn't like strings that look like numbers as struct keys
+//  so we prefix them with an underscore
+global.slot_names = {
+    "_1": { // Infantry
+        "_1": "Left Hand",
+        "_2": "Right Hand",
+        "_3": "Armour",
+        "_4": "Gear",
+        "_5": "Mobility"
+    },
+    "_6": { // Dreadnought
+        "_1": "Left Arm",
+        "_2": "Right Arm",
+        "_5": "Accessory"
+    },
+    "_50": { // Land Raider
+        "_1": "Front",
+        "_2": "Sponson",
+        "_3": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_51": { // Rhino
+        "_1": "Weapon",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_52": { // Predator
+        "_1": "Turret",
+        "_2": "Sponsons",
+        "_3": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    },
+    "_53": { // Land Speeder
+        "_1": "Primary",
+        "_2": "Secondary"
+    },
+    "_54": { // Whirlwind
+        "_1": "Missiles",
+        "_2": "Pintle",
+        "_4": "Upgrade",
+        "_5": "Accessory"
+    }
+};
+
+/// @description This function returns the name of the slot for a given unit type and equipment type.
+/// @param {real} _unit_type - The type of unit to equip, see eUNIT_TYPE.
+/// @param {real} _equipment_type - The type of equipment to equip, see eEQUIPMENT_TYPE.
+/// @returns {string} The name of the slot.
+function get_slot_name(_unit_type, _equipment_type) {
+    var _unit_key_str = $"_{_unit_type}";
+    var _equip_key_str = $"_{_equipment_type}";
+
+    if (global.slot_names[_unit_key_str] && global.slot_names[_unit_key_str][_equip_key_str]) {
+        return global.slot_names[_unit_key_str][_equip_key_str];
+    }
+    return "Unknown";
+}
```

Comment:
Well, yeah. I came to this, which is your suggestion with the switch.
I'm not sure if "eEQUIPMENT_TYPE" enum is nessessary, because slots are just slots. First, second, etc.
But this is just how I think.
```suggestion
/// @description This function returns the name of the slot for a given unit type and equipment type.
/// @param {real} _unit_type - The type of unit to equip, see eUNIT_TYPE.
/// @param {real} _equipment_slot - The equipment slot number.
/// @returns {string} The name of the slot.
function get_slot_name(_unit_type, _equipment_slot) {
    // Map unit types to slot name maps
    switch (_unit_type) {
        case eUNIT_TYPE.Infantry:
            switch (_equipment_slot) {
                case 1: return "Left Hand";
                case 2: return "Right Hand";
                case 3: return "Armour";
                case 4: return "Gear";
                case 5: return "Mobility";
                default: return "Unknown";
            }
        case eUNIT_TYPE.Dreadnought:
            switch (_equipment_slot) {
                case 1: return "Left Arm";
                case 2: return "Right Arm";
                case 5: return "Accessory";
                default: return "Unknown";
            }
        case eUNIT_TYPE.LandRaider:
            switch (_equipment_slot) {
                case 1: return "Front";
                case 2: return "Sponson";
                case 3: return "Pintle";
                case 4: return "Upgrade";
                case 5: return "Accessory";
                default: return "Unknown";
            }
        case eUNIT_TYPE.Rhino:
            switch (_equipment_slot) {
                case 1: return "Weapon";
                case 4: return "Upgrade";
                case 5: return "Accessory";
                default: return "Unknown";
            }
        case eUNIT_TYPE.Predator:
            switch (_equipment_slot) {
                case 1: return "Turret";
                case 2: return "Sponsons";
                case 3: return "Pintle";
                case 4: return "Upgrade";
                case 5: return "Accessory";
                default: return "Unknown";
            }
        case eUNIT_TYPE.LandSpeeder:
            switch (_equipment_slot) {
                case 1: return "Primary";
                case 2: return "Secondary";
                default: return "Unknown";
            }
        case eUNIT_TYPE.Whirlwind:
            switch (_equipment_slot) {
                case 1: return "Missiles";
                case 2: return "Pintle";
                case 4: return "Upgrade";
                case 5: return "Accessory";
                default: return "Unknown";
            }
        default:
            return "Unknown";
    }
    return "Unknown";
}
```

Reply:
There are probably some interesting ways of doing this with `ds_` stuff, but I have 0 experience with them and avoid them, as they are not garbage collected and have to be perfectly coded.